src/bloomfilter.h

```diff
@@ -6,26 +6,29 @@
 #include "mmapbitarray.h"
 #define BF_CURRENT_VERSION 1
 
-struct _BloomFilter {
+typedef struct {
     uint64_t max_num_elem;
     double error_rate;
     uint32_t num_hashes;
-    uint32_t hash_seeds[256];
-    /* All of the bit data is already in here. */
-    MBArray * array;
     unsigned char bf_version;
+
+    uint32_t hash_seeds[256];
```

Comment:
You're moving stuff around. You can't move stuff around without breaking binary compatibility with existing BFs.

While it may make the act of copying fields over a tad more involved, I think you should do this without changing the structure at all.

Reply:
I needed to move some fields around to remove the problematic fields from the header. Adding an `unused` field and modifying the `reserved` array size to compensate for the removed fields I was able to restore the binary compatibility.

Reply:
No, you have to leave the structure alone to be able to restore binary compatibility.

It's not just the size of the struct, it's the location of fields in the struct. These files are memory-mapped and read from file, if you move things around, the earlier BFs are unreadable.

Reply:
Sorry... I was confused. The struct that's stored in file is another struct (MmapBitArray), so this one seems internal and you can play with it however you like.

So... go ahead and destroy binary compatibility if you want ;)

Reply:
Disregard the deleted message. This module is really ugly and unclear about what it does. It seems it steps onto the first few elements of the BloomFilter struct, so you must keep it compatible. But only the first few.

Reply:
Hahaha, either way I modified the `Header` struct to be the same as `_BloomFilter` was (I just renamed the unused fields to make clear that they are now deprecated to avoid potential bugs). I have checked and I can open bloomfilters created with the previous version just fine and the values are as expected. Could you please review it again to make sure that I'm not missing something? Thanks!

Reply:
Ok, yes. There's a lot of black magic involved. I would leave the data structures alone, odds are you'll break something if you don't. This is not clear code.

Reply:
> Hahaha, either way I modified the `Header` struct to be the same as `_BloomFilter` was (I just renamed the unused fields to make clear that they are now deprecated to avoid potential bugs). I have checked and I can open bloomfilters created with the previous version just fine and the values are as expected. Could you please review it again to make sure that I'm not missing something? Thanks!

Yeahno. I'm absolutely sure you don't need to change anything in the data structures to implement this. So if you change something, it's wrong.

You just have to have an option to create 2 independent BFs with the same seeds. That's all.